### ISSUE TITLE
Why is there no losses.backward and optimizer.step on training_steps?

### ISSUE THREAD

**Issue by sarmientoj24** [Erlemar/wheat#1]
I am using this as a baseline on my custom dataset. While I have used Pytorch for FasterRCNN before (not with Pytorch Lightning), I am curious as to why it does not have the typical ending of each training step like the ff:

```
...
losses = sum(loss for loss in loss_dict.values())
optimizer.zero_grad()
losses.backward()
optimizer.step()
...
```

I am not seeing this being implemented explicitly in the code and the utils. Am I missing something why isnt this included?

**Comment by Erlemar** [comment]
I'm using Pytorch Lightning, so I don't have to write all of the training code with `backward()` and other things. In the following article you can see how can Pytorch code be transformed into Pytorch Lightning code.

https://towardsdatascience.com/from-pytorch-to-pytorch-lightning-a-gentle-introduction-b371b7caaf09

Here is the part about training loop:
https://miro.medium.com/max/700/1*sary1TInxeXMLW-44-_RDg.jpeg

**Comment by sarmientoj24** [comment]
I see. Thank you for that. I am currently using it for multi-class detection. Is it correct that if i have 19 classes, it has to be labelled 1-19 but num_classes to the predictor should be 20?

**Comment by Erlemar** [comment]
Yes, you are correct!

**Comment by sarmientoj24** [comment]
I also noticed that you did not have something like 
```
{'class_name': 'albumentations.augmentations.transforms.Normalize', 'params': {'p': 1.0}},
```
From PyTorch docs in training FasterRCNN, they seem to get images normalized by those values on transforms.Normalize. Any particular reason why there isnt such on yours? Is it because you already performed this:

```
image /= 255
```

**Comment by Erlemar** [comment]
Yes, in this repository I did this basic preprocessing as a baseline. Indeed, it would be better to remove this line from the Dataset and add `Normalize` from `albumentations` to config.
